### PR TITLE
feat: Show resoruces per node

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:

- Calculate pod resources per node

**Related issue(s)**
https://github.com/kyma-project/busola/issues/2817

**Definition of done**

- [x] The PR's title starts with one of the following prefixes:
  - feat: A new feature
  - fix: A bug fix
  - docs: Documentation only changes
  - refactor: A code change that neither fixes a bug nor adds a feature
  - test: Adding tests
  - chore: Maintainance changes to the build process or auxiliary tools, libraries, workflows, etc.
- [x] Related issues are linked. To link internal trackers, use the issue IDs like `backlog#4567`
- [x] Explain clearly why you created the PR and what changes it introduces
- [x] All necessary steps are delivered, for example, tests, documentation, merging

### CODE REVIEW

File: src/components/Nodes/NodeResources/NodeResources.js [hunk 1]
```diff
@@ -4,9 +4,10 @@ import { Card, CardHeader } from '@ui5/webcomponents-react';
 import { roundTwoDecimals } from 'shared/utils/helpers';
 import './NodeResources.scss';
 
-export function NodeResources({ metrics }) {
+export function NodeResources({ metrics, resources }) {
   const { t } = useTranslation();
   const { cpu, memory } = metrics || {};
+  console.log(resources);
```
Comment (chriskari) [inline src/components/Nodes/NodeResources/NodeResources.js:10]: remove console log

File: src/resources/Namespaces/ResourcesUsage.js [hunk 2]
```diff
@@ -18,25 +18,44 @@ const MEMORY_SUFFIX_POWER = {
...
+export function getCpus(cpuString) {
+  if (!cpuString || cpuString === '0') {
     return 0;
   }
 
+  const suffix = String(cpuString).slice(-1);
+
+  const suffixPower = CPU_SUFFIX_POWER[suffix];
+  if (!suffixPower) {
+    console.log(cpuString);
```
Comment (chriskari) [inline src/resources/Namespaces/ResourcesUsage.js:54]: remove console log

**Review by chriskari — COMMENTED** [review]
Screenshot 2024-11-12 at 11 41 14

to prevent the chart looking like this you could add the following changes:

adjust the `roundTwoDecimals` function in `helpers` to the following:
`export function roundTwoDecimals(number) {
  return Number(parseFloat(number).toExponential(2));
}`

additionally, in `UI5RadialChart.js` round the percentage using this helper function:
`const percent = max && value ? roundTwoDecimals((value * 100) / max) : 0;`

**Comment by dbadura** [comment]
> additionally, in `UI5RadialChart.js` round the percentage using this helper function:
> 
> ```
> const percent = max && value ? roundTwoDecimals((value * 100) / max) : 0;
> ```

Regarding percentage rounding. I wouldn't do that because it's:
- Current resolution for percentage is precise enought 
- `kubectl describe node` return integers as percent values
- looks uglier

**Review by chriskari — APPROVED** [review]

**Pull request merged**